src/bls12_381/ec.rs

```diff
@@ -166,6 +143,39 @@ macro_rules! curve_impl {
             fn into_projective(&self) -> $projective {
                 (*self).into()
             }
+
+            fn hash(t: Self::Base) -> Self
```

Comment:
> This is injecting into the curve rather than hashing.
Yes, right. 

>  We should also implement the recovery part of Fouque–Joux–Tibouchi.
could you please provide me with a reference? I'm not sure I found the paper you have in mind. 

> I've not checked that the algorithm matches the paper; will do that in another round of review.
ok!

>  Add a reference to the paper in a comment.
done!